azure/http.go

```diff
@@ -159,6 +159,11 @@ func (client *Client) createAzureRequest(url string, requestType string, content
 	var err error
 
 	url = fmt.Sprintf("%s/%s/%s", azureManagementDnsName, client.publishSettings.SubscriptionID, url)
+
+	if client.baseURL != "" {
+		url = fmt.Sprintf("%s/%s/%s", client.baseURL, client.publishSettings.SubscriptionID, url)
```

Comment:
yeah you are adding URL crafting logic to here. this is wrong. all clients must have a `baseURL` – either user-passed or `azureManagementDnsName` and this logic should not be here but at an upper level.

This method should just do c.BaseUrl+"/"+subs+"/+url